### ISSUE TITLE
GeoNode API V2 - Missing information for remote services

### ISSUE THREAD

**Issue by allyoucanmap** [geosolutions-it/nexus-geonode#81]
The API V2 are returning incomplete information in the resources api for remote services

Steps to reproduce:
- create a remote services (eg. with https://master.demo.geonode.org/geoserver/wms)
![image](https://user-images.githubusercontent.com/19175505/115003478-1fc9a300-9ea6-11eb-8ccb-62afbbaa1fcd.png)
- do a request to resource and look for the service  just created http://localhost:8000/api/v2/resources/{pk}

_Current Result_:
- Some information are missing such as detail_url, resource_type, thumbnail_url ... and the new homepage is not able to show to filter and display datails about services (the detail panel crashes the application)

![image](https://user-images.githubusercontent.com/19175505/115003813-7505b480-9ea6-11eb-9471-2a37dd4eb2df.png)

_Expected Result_:
- The service resource should return properties similar to other resources

---

Note:
There is not a specific endpoint for services
![image](https://user-images.githubusercontent.com/19175505/115005100-cfebdb80-9ea7-11eb-8623-2c105ab8800b.png)

**Comment by marthamareal** [comment]
@giohappy  In this Fix should we create a v2 endpoint for services (`/api/v2/services`)?

**Comment by giohappy** [comment]
> @giohappy In this Fix should we create a v2 endpoint for services (/api/v2/services)?

no @marthamareal. I've changed the title of the issue since it was misleading. We're talking about remote layers, not the remote services from which remote layers are obtained.
Remote layers are stored as Layer instances, with the [Layer.remote_service](https://github.com/GeoNode/geonode/blob/190e5f0939c3e89eca1d2dcd16f79cd867fba3a9/geonode/layers/models.py#L191) field populated.

**Comment by giohappy** [comment]
@allyoucanmap it's not clear to me what you're requesting. Why do you need the services endpoint? All the info required to show remote layers should be provided inside the layers/ response. 
Could you clarify?

**Comment by allyoucanmap** [comment]
@giohappy @marthamareal This issue is referring to the services and not remote layers.

In the new homepage the remote services are not displayed correctly in the cards because are missing some information about them in the resources endpoint:

https://master.demo.geonode.org/api/v2/resources/468 this is a remote service in geonode

- missing detail_url (probably it's not needed in the new app structure)
- missing thumbnail_url

In my local machine was missing also the resource_type properties and for this reason I was not able to filter by type
I'll double check later if it's still an issue locally.

**Comment by giohappy** [comment]
@allyoucanmap it's clear. The misunderstanding was the term "services", which is not the remote layer, but the source of the remote layer.
@marthamareal please check why those information isn't populated.

**Comment by marthamareal** [comment]
My proposal solution would be calling resource [resourcebase_post_save](https://github.com/GeoNode/geonode/blob/190e5f0939c3e89eca1d2dcd16f79cd867fba3a9/geonode/base/models.py#L1915)  on a service in [post_save_service](https://github.com/GeoNode/geonode/blob/190e5f0939c3e89eca1d2dcd16f79cd867fba3a9/geonode/services/signals.py#L55).
With this, the service `detail_url` and `thumbnail_url` will be set and returned in the response.

What do you think?

**Comment by giohappy** [comment]
@marthamareal is this the way other types of resources populate those fields?

**Comment by marthamareal** [comment]
Yes. 
[Layers](https://github.com/GeoNode/geonode/blob/190e5f0939c3e89eca1d2dcd16f79cd867fba3a9/geonode/layers/models.py#L755)
[Documents](https://github.com/GeoNode/geonode/blob/190e5f0939c3e89eca1d2dcd16f79cd867fba3a9/geonode/documents/models.py#L270)

**Comment by giohappy** [comment]
ok, please proceed.

**Comment by giohappy** [comment]
Resuming and clarifying.

1 - We don't want to list "remote services" in the card. They are nor resources by themselves. The management of remote services will be done through the new harvesting APIs.

2 - We want to list "remote layers" instead, but from the few things read in the chat it looks like the required information is not always populated. @marthamareal please write where we are with this and report any problem that you're facing

**Comment by marthamareal** [comment]
The reason as to why we were having remote services listed was because they were created with attribute(metadata_only=False)

**Comment by giohappy** [comment]
@marthamareal are you already working to the changes you mentioned above?

**Comment by marthamareal** [comment]
> @marthamareal are you already working to the changes you mentioned above?

Yes I worked on them was waiting for confirmation then i raise the PR.

**Comment by giohappy** [comment]
@afabiani is metadata_only=True for remote services ok for the current remote service management functionalities?

**Comment by afabiani** [comment]
@giohappy yes it should be `True` for the definition of a `Service`, which inherits form ResourceBase.

Of course the `Layers` derived from the former `Service` should still have that field set to `False`.

**Comment by giohappy** [comment]
@marthamareal please proceed
